### DISCUSSION TITLE
Missing upload options

### DISCUSSION THREAD

**Discussion by maxxer** [WeblateOrg/weblate#5577] [General]
Hi.
I'm translating a PHP application written with Yii2, using .po files. I uploaded the first po file and started translating, now I figured out some messages were wrong and I'd like to upload a new source po file with updated strings, but when I go to the Italian translation of the component (I use it as source lang) the only [upload option](https://docs.weblate.org/no/latest/user/files.html#import-methods) I have is the *Update source strings*. The message above the option says I can enable file rewrite but I couldn't find where...

![immagine](https://user-images.githubusercontent.com/240201/109949279-79518600-7cdb-11eb-8330-b6c9b933cea3.png)

thanks

**Comment by nijel** [comment]
It can have several reasons, please pick what might apply to you (not sure if I list all of them):

* You are uploading to the source of bilingual translations
* You don't have permissions
* The component is locked

**Reply by maxxer** [reply]
I am indeed uploading to the source language. 

My purpose is to update the source strings, as I made some fixes in the source files. How do I do that? Thanks

P.S. I remember you for the Nokia stuff :D

**Reply by nijel** [reply]
If you want to update the source strings then uploading POT file here and choosing "Update source strings" is what you most likely want...

**Reply by HarukaMa** [reply]
I'm also having a similar issue here:

My project is being developed offline, so I created a new component "from scratch" and intended to manually upload POT files. 

I tried to upload a simple file:

```pot
msgid "Username is required"
msgstr ""

msgid "Password is required"
msgstr ""

msgid "Login Failed"
msgstr ""

msgid "OK"
msgstr ""

msgid "Username"
msgstr ""

msgid "Password"
msgstr ""

msgid "Login"
msgstr ""

msgid "Register"
msgstr ""
```

I also only have "Update source strings" mode, and uploading this file would only result in "No strings were imported from the uploaded file.".

"Manage strings" settings is disabled by default, and I can't enable it as weblate does not have the file needed:

![image](https://user-images.githubusercontent.com/861659/110768216-5e15e600-829a-11eb-871f-b225756c26df.png)

It's blocking me from saving other settings here.

I tried to push the file to weblate repository, but git returns 403 on push operations.

Edit: After disabling "Adding new translations", now it stuck on:

![image](https://user-images.githubusercontent.com/861659/110790773-ce316580-82b4-11eb-9516-fd87ee4934a7.png)

Of course I don't have this file as I'm not able to use VCS integration, however it makes me unable to edit settings at all.

**Reply by nijel** [reply]
The component created from scratch should have valid configuration before, what was that?

You cannot upload a new pot file to the existing component which does not have it configured. If you want to have it in the Weblate repository create a zip file with desired files and upload it when starting.

**Reply by HarukaMa** [reply]
So for initial configuration I'll have to upload a zip file containing all needed files when creating the component?

When I chose "from scratch" with gettext PO file format, it would create the following default configurations:

![image](https://user-images.githubusercontent.com/861659/110898026-d7b4df00-8341-11eb-9834-5951e66ff4a5.png)
![image](https://user-images.githubusercontent.com/861659/110898049-e00d1a00-8341-11eb-86f8-297401951c44.png)

In this case, there would be no PO files in the weblate repo, therefore the settings form will not be accepted as it would complain no PO files are present. Also I wanted to start from POT files, as I don't have initial PO files to start with. (The app is using bilingual format so it should be fine, right?)

**Reply by nijel** [reply]
The creation from scratch is intended for situations where you built the translations from zero - you add languages and strings in Weblate. If you have some content better to upload that as zip or use full featured VCS integration.
